added puncuation

itolimaesther opened the pull request:
## Describe your changes
I added punctuation(support,) to the sentence "An empty alt attribute has the most wide-spread support so that should be your go-to choice when using icons." for better readability.

## Screenshots - If Any (Optional)


## Link to issue
<!-- Example: Closes #31 -->
Closes #367 

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] Followed the repository's [Contributing Guidelines](/CONTRIBUTING.md).
- [x] I ran the app and tested it locally to verify that it works as expected.
- [x] I have checked my code with an automatic accessibility tool such as Axe Dev Tools or Wave 
      and it shows no errors.
## Additional Information (Optional)
Any additional information that you want to give us.

itolimaesther commented:
@EmmaDawsonDev please review this PR.

EmmaDawsonDev reviewed (changes requested):
Thanks for your PR Esther. For some reason your code formatter has formatted some of the code snippets which makes them display wrongly. Please remove the formatting so that only the punctuation change is left.

itolimaesther commented:
Alright.

itolimaesther commented:
@EmmaDawsonDev will I need to create another PR?

EmmaDawsonDev commented:
No, you should just be able to make the changes and push them and they will be reflected here.

itolimaesther commented:
I have made some code changes, removed the prettier formatter, and also added content for the link. Please check and let me know if the issue persists.

EmmaDawsonDev commented on components/ContentTemplates/LinksTemplate.tsx:
```diff
@@ -1,5 +1,392 @@
+
+				<CodeBlock
+					codeSnippet={`<a href="https://www.freecodecamp.org">Read more
+	<span class="sr-only">Read more about FreeCodeCamp</span>
```
On the buttons page the same class is called .visually-hidden as this technique is not always used only for screen readers. It would be great if you could update it here to use the same naming conventions

EmmaDawsonDev commented on components/ContentTemplates/LinksTemplate.tsx:
```diff
@@ -1,5 +1,392 @@
+			</TemplateSection>
+			<TemplateSection sectionName="linkText" title="Descriptive Link Text">
+				<p>
+					Link text is the text that describes a hyperlink on a webpage. It's
+					important to have descriptive link text to provide clear context for
+					screen readers and all users.
```
```suggestion
					Link text is the text between the opening anc closing anchor tag that describes a hyperlink on a webpage. It's important to have descriptive link text to provide clear context for all users about where the link leads.
```

EmmaDawsonDev commented on components/ContentTemplates/LinksTemplate.tsx:
```diff
@@ -1,5 +1,392 @@
+				</p>
+
+				<p>
+					However, ARIA labels may not provide perfect accessibility. A better
+					solution is to use a <strong>{`${"<span>"}`}</strong> element within
+					the <strong>{`${"<a>"}`}</strong> element to provide descriptive text
+					that's accessible to screen readers but not visible to sighted users{" "}
```
Can you explain more about why a hidden span is more suitable than aria-label?

EmmaDawsonDev replied:
```suggestion
					that's accessible to screen readers but not visible to sighted users.
```

EmmaDawsonDev commented on components/ContentTemplates/LinksTemplate.tsx:
```diff
@@ -1,5 +1,392 @@
+					languageType={"css"}
+				/>
+				<p>
+					<a href="https://www.freecodecamp.org">
+						Read more
+						<span className={styles.srOnly}>Read more about FreeCodeCamp</span>
+					</a>
```
I don't think we need the actual example

EmmaDawsonDev commented on components/ContentTemplates/LinksTemplate.tsx:
```diff
@@ -1,5 +1,392 @@
+					Avoid using vague phrases like "Click Here" or "Read More."
+				</p>
+				<p>
+					If you must use <strong>non-descriptive text</strong>, you can make it
+					more accessible with ARIA attributes for screen readers.
```
It would be good to give an example of where this might be the case eg a list of cards

EmmaDawsonDev commented on components/ContentTemplates/LinksTemplate.tsx:
```diff
@@ -1,5 +1,392 @@
+						universal convention that helps all users understand what's
+						clickable.
+						<CodeBlock
+							codeSnippet={`a {
+    text-decoration: underline;
+}`}
+							languageType={"css"}
+						/>
+					</li>
+					<li>
+						<strong>Ensure Sufficient Color Contrast: </strong>
+						Make sure the color of your links contrasts well with the background
+						color. This ensures that people with visual impairments can easily
+						distinguish the links from the surrounding text.
+					</li>
+					<li>
+						<strong>Add Link State on Click or Visit: </strong>
+						To aid users in identifying which links they've already visited or
+						clicked, consider adding a visual change, such as making the link
+						text bold or giving it a bright outline color.
+						<CodeBlock
+							codeSnippet={`a:visited {
+    font-weight: bold;
+    outline: 2px solid #00f; /* Use a bright color for the outline */
+}`}
+							languageType={"css"}
```
I think it would be best to move the code blocks outside of the list and combine them into one

EmmaDawsonDev commented on components/ContentTemplates/LinksTemplate.tsx:
```diff
@@ -1,5 +1,392 @@
+						<strong>{`${"<a>"}`}</strong> to create a clickable link. This makes
+						it clear that the image is interactive and leads to another page.
+						<CodeBlock
+							codeSnippet={`<a href="#">
+    <img alt="Old brown paper texture" src="/images/imagesTemplate/oldPaperTexture.jpg" width="300" height="200"/>
+</a>`}
```
Again, probably best to move the code blocks outside of the list and combine it into one.

EmmaDawsonDev reviewed (changes requested):
This is looking really good so far. I've just left a few comments for minor things that need improving

itolimaesther commented:
Alright. I have seen all your comments. I make the corrections and I will make a push again.

EmmaDawsonDev commented:
I'm going to merge this into a temporary branch and fix the issues when I get time. Thanks so much for your help so far :)

itolimaesther commented:
I apologize for finishing with this. I have been down for some time. Please let me know if there are anything I need to add.

EmmaDawsonDev commented:
No worries. I fully understand that life gets in the way sometimes.
